### PR TITLE
Feat add checkout page and component placeholders

### LINKED ISSUE #167: Checkout Page Component
# Checkout Page Component
## Overview / Summary
Create a Checkout page component
This component should display the Checkout Nav, an optional "TopHat" component, a Checkout Cart Review, and a Checkout Actions Component
This component should be added to the Storefront starter-kit

## Expected use cases
This component will be used any time a customer navigates to the checkout.

### Where should this component live
This component should be created in the starterkit.

## Images
image

```jsx
// Pseudo-markup for the mobile cart Popover
<Page>
  <div name="topHat">TopHat Message</div>
  <CheckoutNav />
  <CheckoutActions />
  <CheckoutCartReview />
</Page>
```

### Props
```js
propTypes: {
  cart: PropTypes.shape(),
}
```

### Data Fetching
None. A cart decorator or HOC will later be wrapped around this component to do the GraphQL cart query and pass in the cart. For now, mock the incoming cart prop in defaultProps.

### State Store
All data comes from props. No global state used.

### Routing / Query String
This page is rendered on the `/cart/checkout` route

### Render Criteria
The checkout should be rendered when a customer has a valid, unexpired cart with at least one item in it and visits the checkout route

### Breakpoints
The Mobile view should be shown on browsers 959px and narrower, the desktop view should be shown from 960px and wider.

#### Normal
Zeplin:
Guest: https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f4fa90d3b184213318da
Account: https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f470eefa9c371dd3c488

#### Empty
This component should not be rendered without at least one item in the cart. Render the empty cart message instead.

#### Normal
Zeplin:
Guest: https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f55b5650607e22fdbaf6
Cart review open: https://app.zeplin.io/project/5afc82f5e8d73e250ff9d855/screen/5b02f55be2af3f6a21b4200a

#### Empty
This component should not be rendered without at least one item in the cart. Render the empty cart message instead.

### PR DESCRIPTION
**Pull request by willopez**
Resolves #167
Impact: Minor
Type: Feature

## Feature
Add checkout page component and corresponding route, placeholders are used for component not yet built.

### CODE REVIEW

File: src/pages/cart.js [hunk 1]
```diff
@@ -58,10 +60,17 @@ const styles = (theme) => ({
...
 @trackProductViewed()
+@inject("uiStore")
+@observer
 @withStyles(styles)
 class CartPage extends Component {
   static propTypes = {
-    classes: PropTypes.object
+    classes: PropTypes.object,
+    shop: PropTypes.shape({
+      currency: PropTypes.shape({
```
Comment (kieckhafer) [inline src/pages/cart.js]: Why do we include this `currency` shape here? We don't seem to use it anywhere.

We do use `name` and `description` under `shop` though, should those fields be here instead?

File: src/pages/checkout.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
 
-@inject("routingStore", "uiStore")
 @observer
 class Checkout extends Component {
   static propTypes = {
+    classes: PropTypes.object,
     shop: PropTypes.shape({
       currency: PropTypes.shape({
```
Comment (kieckhafer) [inline src/pages/checkout.js]: same question as above.

**Comment by willopez** [comment]
@kieckhafer I have addressed your comments

**Comment by willopez** [comment]
@kieckhafer @machikoyasuda waiting for the next release of component library for styles  to match design

**Comment by willopez** [comment]
@kieckhafer this is ready for review. I spoke with @rymorgan regarding some new updates to the checkout page designs that will be in implemented soon. I'll create another issue to match the new designs updates once they are available.

**Comment by willopez** [comment]
Issue related upcoming updates to the checkout top hat: https://github.com/reactioncommerce/reaction-next-starterkit/issues/180

**Comment by machikoyasuda** [comment]
I got it running!

This is the way I compared the mock with the page:

screen shot 2018-07-19 at 4 42 48 pm

I used the Zeplin desktop app, and clicked the [Pop Out option](https://support.zeplin.io/zeplin-101/comparing-appwebsite-with-designs-using-pop-out) to get a 100% scaled view of the page, and compared it with the browser's `http://localhost:4000/cart/checkout`.

I noticed one thing, aside from https://github.com/reactioncommerce/reaction-next-starterkit/issues/180 - which you already have as an issue:

screen shot 2018-07-19 at 4 49 09 pm

The two components should have a whitespace margin around them. I tried changing some props of the grid in console and got something like, with the first component at a grid value of 5 and the other with 3, and adding flex-box centering:

screen shot 2018-07-19 at 5 13 59 pm

It's not _exact_ but it will help us get closer to it  -- cc @rymorgan

**Comment by kieckhafer** [comment]
@willopez there is a `layout.mainContentMaxWidth` var inside the theme that you can use to create a wrapper.

**Pull request merged**